src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -181,39 +168,29 @@ public PriorityScheduler(int corePoolSize, int maxPoolSize,
   public PriorityScheduler(int corePoolSize, int maxPoolSize,
                            long keepAliveTimeInMs, TaskPriority defaultPriority, 
                            long maxWaitForLowPriorityInMs, ThreadFactory threadFactory) {
-    ArgumentVerifier.assertGreaterThanZero(corePoolSize, "corePoolSize");
-    if (maxPoolSize < corePoolSize) {
-      throw new IllegalArgumentException("maxPoolSize must be >= corePoolSize");
-    }
-    
-    //calls to verify and set values
-    setKeepAliveTime(keepAliveTimeInMs);
-    setMaxWaitForLowPriority(maxWaitForLowPriorityInMs);
-    
+    this(new WorkerPool(threadFactory, corePoolSize, maxPoolSize, 
```

Comment:
The shared memory in "PriorityScheduler" was really the workers.  Of all the changes I think this one is the cleanest.

You will see in the other classes the new static classes were really more of structs, just holding datastructures which we reference and call into with the same functionality.

However in this class I actually duplicated the functions and moved the functionality into this inner class (more object orriented).

@lwahlmeier  I want your opinion on which is cleaner, should I duplicate the functions in the other classes as well?  I do think that's cleaner, but it also increases the jar size.